### ISSUE TITLE
Normalize theme/topic/class throughout interface

### ISSUE THREAD

**Issue by mmisw** [mmisw/mmiorr#99]
_From [grayb...@mbari.org](https://code.google.com/u/109634240660495836000/) on March 13, 2009 12:17:54_

What steps will reproduce the problem? 1. User creates ontology via voc2rdf
2. User registers ontology
3. User updates metadata for ontology What is the expected output? We need to be consistent and clear in how we refer to the primary concept of the ontology (in a 
way that allows for multiple classes for terms within the ontology) What do you see instead? Different terms used for the same concept, and the application of those terms are unclear to the 
user. Please use labels and text to provide additional information. Will add more thoughts later.

_Original issue: http://code.google.com/p/mmisw/issues/detail?id=99_

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 13, 2009 15:15:22_

Agree. This is actually closely related with https://code.google.com/p/mmisw/issues/detail?id=69 : "new shortName doesn't replace
previous shortName (resource type)." We need to clarify the relationship between the
"short name" (aka resource type), which is used to construct the URI for the
ontology, and the actual class (and eventually multiple classes and other entities
that can be supported in the future by voc2rdf) contained in the ontology.

[note: I'm deprecating the org.mmisw.* labels; please use the corresponding simple
labels instead, for example, "voc2rdf" instead of "org.mmisw.voc2rdf." --thx.]

**Cc:** carueda  
  **Labels:** -org.mmisw.voc2rdf -org.mmisw.ont voc2rdf ont ontmd

**Comment by mmisw** [comment]
_From [grayb...@mbari.org](https://code.google.com/u/109634240660495836000/) on March 19, 2009 11:19:34_

See also `#70` (creating instructions).

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 19, 2009 18:20:39_

Issue 70 has been merged into this issue.

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 20, 2009 11:41:06_

What about the following approach:

Fields involved:

- (rt) (for "resource type"): a 'resource type' field, basically as currently
included in OntMd, but also to be applied in voc2rdf as proposed in this approach. In
particular, I propose that the value for this field be strictly selected from the
ResourceType vocabulary ( http://mmisw.org/ont/mmi/resourcetype ). (We would add a
special 'Pending' resource type for the case where no other option is satisfactory;
then use some mechanism to "officially" include the new resource type).

- (sn) (for "short name"): a 'short name' field where the user can specify the
desired shortName component of the final URI for the ontology. This can be
initialized according to the value entered in (rt), but --from the tool's
perspective-- is text only subjected to the appropriate character set. Of course,
associated tooltips will encourage certain patterns and include links to MMI.

- (cn) (for "class name"): In voc2rdf, this is the name of the OWL class for the
terms. It can also be initialized according to (rt). (BTW, the description of this
issue seems to suggest that voc2rdf should support multiple classes, right? If so, we
need to create a specific issue and give more details for that.)

Please comment.   -thx.

**Owner:** carueda  
  **Cc:** -carueda grayb...@mbari.org paul.r.alexander bermud sara.m.haines

**Comment by mmisw** [comment]
_From [ber...@gmail.com](https://code.google.com/u/105051227815968870037/) on March 20, 2009 12:27:23_

"resource type" = super class at MMI
"class name" - user class and it is a subclass of "resource type"
"short name" - individual of class "class name" it could also be subclass of "class
name" if we want to create classes and not individuals

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 20, 2009 12:44:27_

Thanks Luis. Some comments re what you say:

"class name": If this *always* has to be a subclass of "resource type", then we
should probably, for consistency pruposes, organize the resourcetype ontology in a
hierarchy of classes. But, perhaps other relationship could also be possible between
"class name" and "resource type," right?

"short name": Note: this is merely for constructing the URI for the ontology. And
this piece shouldn't be forced to say anything about any particular instance of the
"class name" class.

**Comment by mmisw** [comment]
_From [ber...@gmail.com](https://code.google.com/u/105051227815968870037/) on March 20, 2009 13:20:19_

| "class name": If this *always* has to be a subclass of "resource type"
not always but lets start with of we encounter other possible relations then sure. 

|"short name"
yes you are right I got confused. I thought it was the short name of a term when
using voc2rdf

**Comment by mmisw** [comment]
_From [grayb...@mbari.org](https://code.google.com/u/109634240660495836000/) on March 20, 2009 13:49:58_

Luis' description of 'short name' refers to the terms, not to the URI 'base concept' that Carlos was originally discussing.  (I know Carlos realizes this, but 
I'm not sure Luis does.)

So Carlos is saying my resourceType can be 'parameter', my class name can be 'standard name' but 'standard name' is a subclass of 'parameter', and the 
URL can use 'stdname', as in mmisw.org/ont/cf/stdname/water_temperature.

As this points out, the class name may not fit under any of the existing resource types. 

On the other hand, I don't like constraining resource types to one of ours.  Why can't it be any semantic concept?  I do like giving them a selection of 
preferred resource types, but allowing them to specify their own (as long as it is a concept associated with an ontology or URI) seems reasonable.

I like the idea of Voc2RDF and OntMD using the same ResourceType.

I like the idea of class name defaulting to Resource Type. But I think the class name should be allowed to be anything.  For example: My resourceType is 
'parameter'.  but my class name might be 'AirTemperatureParameter', or 'OOSTethys_Viewer_Parameter', or even 'John Set1 Parameter' (I forget what the 
syntax rules for the class name are, sorry!).  This name is where I can distinguish my parameter types, and I think it is useful if people do so, rather than 
having 500 'parameter' classes in our system.

Now, what should the URL short name for this concept default to if resourceType != class name? (Remember, the defaults will give resourceType = class 
name = URL short name.) I think it should default to the class name (if one class), or the ResourceType (if multiple classes).  

Note that if Resource Type != class name, the resource type is only used for classification of this ontology, and will not be prominently visible anywhere. 
This approach makes ResourceType into a sorting mechanism, not the method by which the ontology is named. 

A possible point of confusion that this flexibility allows is that I look in the URL /mmiexample/type and I see a class called deviceType (which is 
categorized as Resource Type sensor type). Even more awkward, someone else then wants to use class type, which is fine, but their URL can't be 'type', 
so they use 'type1'.  Now /mmiexample/type1 includes class type, and /mmiexample/type points to deviceType. Very non-intutive; there is no 
guaranteed association of URL to class names any more.  

I could even have multiple URLs all of which link to (different) ontologies for the class 'type', right?  Because the class name is no longer the unique 
identifier for the vocabulary, only the URL is. 

So even more strange, my term 'sampler' in the class 'deviceType' will appear as /mmiexample/type/deviceType, and my term 'typewriter' in the class 
'type' will appear as /mmiexample/type1/typewriter.  

I think the point of the long story is that people should think very carefully about whether they really want a different URL and class type; usually this 
should be discouraged.

I addressed multiple classes in a separate issue (#114), and also entered something for making all terms into a class (#113).

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 20, 2009 15:48:41_

Issue 69 has been merged into this issue.

**Cc:** carueda

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 20, 2009 21:05:20_

attached: whiteboard for voc2rdf plans

**Attachment:** [voc2rdf.jpg](http://code.google.com/p/mmisw/issues/detail?id=99#hc10)

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 21, 2009 18:04:47_

Hmmm, assuming that we want voc2rdf to *also* be useful independently of the MMI
repository (as it has been the case so far),  I now propose the following:

- Allow the user to specify, in a single field, the complete URI for the ontology.
That is, no multiple fields (base URI, authority, shortName) for composing that URI.
Rationale:
  * simpler UI (I emphasize: "for the user")
  * we shouldn't force the user to follow any particular URI structure at this point
(see also how this is done in Protege when you start a new ontology).
  * and note that, if the user decides to submit the generated ontology to our
repository, then OntMd will impose the required structure for the final ontology URI
as it already does.

In summary, the proposed fields for the main voc2rdf interface are:

- "ontology URI" as proposed above;

- "class": the user can choose this class from the terms in http://mmisw.org/ont/mmi/resourcetype.html (as it happens now in OntMd) but can enter
any text for this field. 

- "URI of the class": similar to the current "URI of resource type" in Ontmd (this
will be used by ontmd in case of submission).

- "terms": the usual textarea for the CSV contents.

**Status:** Started  
  **Cc:** -paul.r.alexander -sara.m.haines -carueda

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 21, 2009 19:58:07_

I have implemented my latest proposal so it's easier to evaluate it and counter-propose. http://mmisw.org/ontmd/voc2rdf/ MMI Voc2RDF 1.1.0.beta (20090321194307)

Again, note that the class name and its URI are captured in the metadata fields
omv:acronym and omvmmi:shortNameUri as done in ontmd with the resource type and
corresponding URI, respectively. If you proceed with submission of the generated
ontology, you should see those fields initialized with the values captured in voc2rdf.

**Cc:** paul.r.alexander

**Comment by mmisw** [comment]
_From [grayb...@mbari.org](https://code.google.com/u/109634240660495836000/) on March 22, 2009 09:49:35_

The problem with the approach in Comment 11 is that the average user (by which I mean 95% of our target 
users) will have no idea how to fill out the URL.  And of the ones who do know how they should fill it out, 
most *won't* follow our guidance if we leave it wide open for them. And then, many of those will be confused 
when very odd fields get put into 

Let's be clear that the primary purpose of voc2rdf is to feed the ontology repository. And so it should be an 
experience that aligns with the ontology repository experience, and should serve as training in the 
fundamental principles.  The two user interfaces should map very closely to each other, and be consistent in 
their operation and the meaning of the form entries. 

That said, I don't have a problem creating an *option* for the user to do a different URL, but it should not be 
prominent, indeed should not be noticed by most users.  I don't have any problem supporting general use, 
but really I am trying to get everyone to follow a common practice.

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 26, 2009 16:27:41_

**Labels:** Milestone-Release1.2

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on March 29, 2009 21:49:26_

Voc2RDF overhauled. http://mmisw.org/ontmd/voc2rdf/ MMI Voc2RDF 1.2.0.beta1 (20090329213042)

- More metadata captured (title, creator, description, authority)
- Button convert to RDF now opens a dialog (No more a tab for the conversion)
- class & terms now in a tab (in preparation for multiple classes)
		
- New editable, spreadsheet-like table for the terms:
    - The editable table is now the main interface for capturing the terms
    - A new CSV button displays the current table in CSV format; the user
      can use the text area to update the contents and re-import into the table.

NOTE:
 - Still needs some built-in documentation.
 - No resouceType section yet in the main header of the interface. Relationship
between this resourceType and the class in the vocabulary still unclear. See also issue `#114`

**Cc:** -bermud -paul.r.alexander

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on April 08, 2009 19:01:30_

See also related issue `#84` just implemented.

**Cc:** -grayb...@mbari.org

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on August 31, 2009 14:20:50_

I'm lowering this to "high" as I'm doing several related updates to the various
interfaces (voc2rdf, vine, external ontology, etc.).

**Labels:** -Priority-Critical -Milestone-Release1.2 Priority-High vine portal

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on October 07, 2009 12:46:07_

Now, 'critical' again, as this should be solved to better address the various
adjustments related with the hosting options intended to be provided )flly hosted,
re-hosted, indexed).

Some of the likely adjustments are:

1) omv:acronym NOT used anymore to capture the resource type

2) new omvmmi:hasResourceType to capture the resource type

3) omv:acronym to be used to capture "A short name by which an ontology is formally
known" (see OMV documenation).

4) "short name" used to compose the ontology URI (in the fully hosted mode) NOT to be
captured in any metadata attribute--it's just for purposes of creating the URI.

**Labels:** -Priority-High Priority-Critical

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on November 03, 2009 10:00:20_

Implemented several changes to should address this issue for the most part.
MMI Portal 1.6.0.alpha (20091103085316)

Marking as fixed so more comprehensive testing can continue.  
I suggest to create new more specific entries in the tracker (instead of re-opening
this one) if associated issues are still present.

**Status:** Fixed

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on November 03, 2009 10:05:41_

To summarize the changes related with this issue:

1) omv:acronym NOT used anymore to capture the resource type

2) omvmmi:hasResourceType used now to capture the resource type

3) omv:acronym now used to capture "A short name by which an ontology is formally
known" (consistent with OMV).

4) A "short name" field is captured only for purposes of creating the URI in the
fully hosted mode.

5) For "self-created" ontologies (that is, using the voc2rdf or vine modules), the
omv:acronym is *also* used to compose the URI of the ontology.

**Cc:** grayb...@marinemetadata.org steph_wa...@consolidated.net
